Fix booting issue with FVP BaseR v8R (version >= 11.16.16)

povergoing opened the pull request:
In the [Armv8R AArch64 profile](https://developer.arm.com/documentation/ddi0600/latest/), the Armv8R AArch64 is always in secure mode. But the FVP_BaseR_AEMv8R before version 11.16.16 doesn't strictly follow this rule. It still has some non-secure registers(e.g. CNTHP_CTL_EL2).

The issue is:
Since version 11.16.16, the FVP_BaseR_AEMv8R has fixed this issue. The CNTHP_XXX_EL2 registers have been changed to CNTHPS_XXX_EL2. So the FVP_BaseR_AEMv8R (version >= 11.16.16) cannot boot Zephyr. 

This PR will fix it:
Also means, after the fix, the minimum required version of FVP will be 11.16.16. Since there is almost no one uses the version less than 11.16.16 and also no way to download the older versions, I think I will not spend time doing the logic of the fvp version checking. It's enough to add an announcement of the minimum required FVP version in the board doc

povergoing committed:
arm64: Fix booting issue with FVP V8R >= 11.16.16

In the Armv8R AArch64 profile[1], the Armv8R AArch64 is always in secure
mode. But the FVP_BaseR_AEMv8R before version 11.16.16 doesn't strictly
follow this rule. It still has some non-secure registers
(e.g. CNTHP_CTL_EL2).

Since version 11.16.16, the FVP_BaseR_AEMv8R has fixed this issue. The
CNTHP_XXX_EL2 registers have been changed to CNTHPS_XXX_EL2. So the
FVP_BaseR_AEMv8R (version >= 11.16.16) cannot boot Zephyr. This patch
will fix it.

[1] https://developer.arm.com/documentation/ddi0600/latest/

Signed-off-by: Jaxson Han <jaxson.han@arm.com>
Change-Id: If986f34dc080ae7a8b226bba589b6fe616a4260b

povergoing committed:
board: arm64: fvp_baser_aemv8r: Update the version requirement

The current minimum required version of "Armv8-R AEM FVP" is 11.16.16.

Signed-off-by: Jaxson Han <jaxson.han@arm.com>
Change-Id: Iaaf1ec7fd432753371b58d13fdd29b1278f6c997

carlocaione commented on boards/arm64/fvp_baser_aemv8r/doc/index.rst:
```diff
@@ -27,6 +27,8 @@ To Run the Fixed Virtual Platform simulation tool you must download "Armv8-R AEM
 FVP" from Arm developer [1]_ (This might require the user to register) and
 install it on your host PC.
 
+The current minimum required version of "Armv8-R AEM FVP" is 11.16.16.
```
is it possible to check this at run-time and print an error message instead?

povergoing replied:
Sad, I have tried several ways, no one does really work. I appreciate it if some CMake experts can tell me how to do the check.

carlocaione replied:
@tejlmand ? :)

tejlmand replied:
what you can do in CMake is something similar to this:
```cmake
execute_process(COMMAND ${ARMFVP} --version OUTPUT_VARIABLE out)
string(REPLACE "\n" ";" out ${out})

foreach(line ${out})
  if("${line}" MATCHES "Fast Models[^0-9]*([0-9][0-9]\.[0-9][0-9]\.[0-9][0-9])")
    set(armvfp_version ${CMAKE_MATCH_1})
  endif()
endforeach()

if(armvfp_version VERSION_LESS 11.16.16)
  message(WARNING "<Good warning description here>")
endif()
```
Please checkup if the match format applies to all the variants of arm VFP in use, as well as `\n` replace pattern on windows.

povergoing replied:
@tejlmand Thank you so much. I have tried and it works! But I still have a question, this warning only prints in compile/build stage right? So, that is, after compiling/building, it cannot print check error when we use `west build -t run` with a wrong version FVP? @carlocaione do you think this will be OK?

povergoing replied:
I've updated the PR. If the FVP version not met the requirements, it will show:
```
-- Found assembler: /home/jaxhan01/.local/zephyr-sdk/aarch64-zephyr-elf/bin/aarch64-zephyr-elf-gcc
CMake Warning at /home/jaxhan01/src/zephyrproject/zephyr/cmake/emu/armfvp.cmake:22 (message):
  Found FVP version is "11.16.16", the minimum required by the current board
  is "12.16.16".
Call Stack (most recent call first):
  /home/jaxhan01/src/zephyrproject/zephyr/CMakeLists.txt:1721 (include)
```

tejlmand replied:
> But I still have a question, this warning only prints in compile/build stage right?

@povergoing  This warning will only print during the CMake configure stage.

If you want to print an error on `west build -t run` then you will need to update the `run` target to print a warning instead of trying to execute, for example something like:
```
set(armfvp_warning_message "Found FVP version is \"${armfvp_version}\", "
      "the minimum required by the current board is \"${ARMFVP_MIN_VERSION}\".")
message(WARNING ${armfvp_warning_message})
set(ARMFVP ${CMAKE_COMMAND} -E echo ${armfvp_warning_message}
     COMMAND ${CMAKE_COMMAND} -E false
)
```

at this location: https://github.com/zephyrproject-rtos/zephyr/blob/c5cf46a6703ee805c98191f3985e253bb088f578/cmake/emu/armfvp.cmake#L21-L24

If implementing this error run-time, then please consider the handling of `ARMFVP_FLAGS `:
https://github.com/zephyrproject-rtos/zephyr/blob/c5cf46a6703ee805c98191f3985e253bb088f578/cmake/emu/armfvp.cmake#L42 


Note, a late arrival review comment, this:
https://github.com/zephyrproject-rtos/zephyr/blob/c5cf46a6703ee805c98191f3985e253bb088f578/cmake/emu/armfvp.cmake#L13
should be:
```
if (ARMFVP AND (DEFINED ARMFVP_MIN_VERSION))
```
> False if the constant is 0, OFF, NO, FALSE, N, IGNORE, NOTFOUND, the empty string, or ends in the suffix -NOTFOUND.

povergoing replied:
@tejlmand Really appreciate it, I will fix it in another PR.

povergoing committed:
cmake: armfvp: Add FVP min version check

After the fix of FVP_BaseR_AEMv8R booting issue, the minimum required
version of FVP will be 11.16.16. Add an FVP minimal required version
check in building time.

When the ARMFVP_MIN_VERSION is set in board cmake file, the version
check will be enabled and print a warning.

Signed-off-by: Jaxson Han <jaxson.han@arm.com>
Change-Id: Ibbade0c328b5e91b8830fb35cba6917f08aabbda

povergoing committed:
board: arm64: fvp_baser_aemv8r_smp: Increase CONFIG_MAX_THREAD_BYTES

Increase CONFIG_MAX_THREAD_BYTES to 3 to fix the build issue on
tests/drivers/build_all/modem/ test case.

Signed-off-by: Jaxson Han <jaxson.han@arm.com>
Change-Id: I6a26955bdd7e8b176894fa8246aec63a3a3db05f
